server/subscribe.go

```diff
@@ -466,7 +466,8 @@ func (p *Plugin) getSortedSubscriptions(instanceID types.ID) ([]SubsGroupedByTea
 
 			channel, appErr := p.API.GetChannel(channelID)
 			if appErr != nil {
-				return nil, errors.New("failed to get channel")
+				p.API.LogWarn("getSortedSubscriptions: failed to get channel.", "channelID", channelID, "error", appErr)
```

Comment:
I think this should be logged as DEBUG or INFO. In general WARNING is used for "not as critical as an error, but admin should take action if they want to fix it." This is not a case where the admin can fix the issue.

I'm thinking DEBUG log would be best because it could get a bit spammy in some cases, and there's no way for the admin to remedy that situation on their own. Either way, I know this in draft so it doesn't matter as much since we are going to take care of cleaning them up when appropriate, but 2/5 WARNING doesn't seem like the right choice.

Reply:
I agree.  Right now I was putting in draft, but also seeing how webhooks reacts with the deleted channel.

I'm also in agreeance that DEBUG is likely the best choice

Reply:
@mickmister, note the differences in log messages from the `WebhookWorker` vs `/jira subscribe list`.  The WebhookWorker is declaring the message as an Error vs Debug. I was wondering if we should change the webhook message to debug also, but actually think Error is fine.  A webhook is a reactive action vs. `/jira subscribe list`

Reply:
I don't think the webhook handler should be logging it as Error, because:

- Webhooks are called very often, and this would flood the logs with errors
- The admin can't do anything to remedy the situation, so the errors are pretty useless to have in the logs

I'm thinking DEBUG is appropriate for both the command and webhooks

Reply:
My thought on the webhook remaining an error is so that a webhook error would get seen in the log. Otherwise, a customer would only see the webhook error if they are logging DEBUG messages. I do agree, once an error starts occurring, it would likely flood the logs. 

I'll change to DEBUG

Reply:
> Otherwise, a customer would only see the webhook error if they are logging DEBUG messages

The problem here is that the admin has no way of remedying the situation. Let's say they notice the issue in their logs, what do they do next? Whether it is an error or debug log, it's not useful for admin to know this is happening. 2/5 the webhook code should clean up the issue when it realizes that "100% this channel has been deleted" case.

Anyway, this is out of scope of this PR. Just want to discuss the right thing to do in the other case. If we are cleaning up the data automatically, maybe we should use Warning as the log level to make it surface in the logs. The "warning" is that we are modifying their subscriptions to agree with the deleted channel state.